README.md
# ree
Plugin para Jeedom para la consulta diaria de los precios de las tarifas de la  luz de Red Eléctrica Española

Este plugin nos permitirá consultar de forma diaria las tarifas PVPC españolas, cuyos precios del día siguiente los da REE a través
la API esios normalmente antes de las 22 horas de cada día.

Trás esto, podemos activar el plugin en Jeedom, dentro de la configuración del mismo necesitamos configurar el token personal que tenemos
que solicitar a REE en la dirección consultasios@ree.es, la respuesta suele ser bastante rápida.

Por último, ya solo nos queda crear nuestro equipo y elegir de que tarifa queremos que muestre los datos.
- PVPC: Tarifa normal de precio voluntario pequeño consumidor.
- PVPC 2.0 DHA: Tarifa con discriminación horaria de precio voluntario pequeño consumidor.
- PVPC 2.0 DHS: Tarifa cn discriminación horaria coche eléctrico de precio voluntario pequeño consumidor.

Se aceptan donaciones en https://www.paypal.me/fmolerot


changelog.md
# ree
- 0.3 
  - Añadimos un comando llamado graph al que si le asignamos un nuevo widget que también hemos creado nos genera una gráfica con los precios. El widget se tiene que copia en nuestro jeedom y el fichero se llama PLuz.html. 
- 0.2
  - Añadimos la opción de elegir a que hora quieres que haga el cambio de día. Desde las 8:30 de la tarde, hasta las 12 de la noche. Por defecto, todos van a cambiar a las 23 hora.
  
- Versión 0.1 Incluye todas las tarifas españolas del mercado regulado PVPC.



desktop/js/ree.js

/* This file is part of Jeedom.
 *
 * Jeedom is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jeedom is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jeedom. If not, see <http://www.gnu.org/licenses/>.
 */


$("#table_cmd").sortable({axis: "y", cursor: "move", items: ".cmd", placeholder: "ui-state-highlight", tolerance: "intersect", forcePlaceholderSize: true});
/*
 * Fonction pour l'ajout de commande, appellé automatiquement par plugin.template
 */
function addCmdToTable(_cmd) {
    if (!isset(_cmd)) {
        var _cmd = {configuration: {}};
    }
    if (!isset(_cmd.configuration)) {
        _cmd.configuration = {};
    }
    var tr = '<tr class="cmd" data-cmd_id="' + init(_cmd.id) + '">';
    tr += '<td>';
    tr += '<span class="cmdAttr" data-l1key="id" style="display:none;"></span>';
    tr += '<input class="cmdAttr form-control input-sm" data-l1key="name" style="width : 140px;" placeholder="{{Nom}}">';
    tr += '</td>';
    tr += '<td>';
    tr += '<span class="type" type="' + init(_cmd.type) + '">' + jeedom.cmd.availableType() + '</span>';
    tr += '<span class="subType" subType="' + init(_cmd.subType) + '"></span>';
    tr += '</td>';
    tr += '<td>';
    tr += '<span><input type="checkbox" class="cmdAttr checkbox-inline" data-l1key="isHistorized" /> {{Historiser}}<br/></span>';
    tr += '<span><input type="checkbox" class="cmdAttr checkbox-inline" data-l1key="isVisible" /> {{Affichage}}<br/></span>';
    tr += '</td>';
    tr += '<td>';
    if (is_numeric(_cmd.id)) {
        tr += '<a class="btn btn-default btn-xs cmdAction" data-action="configure"><i class="fa fa-cogs"></i></a> ';
        tr += '<a class="btn btn-default btn-xs cmdAction" data-action="test"><i class="fa fa-rss"></i> {{Tester}}</a>';
    }
    tr += '<i class="fa fa-minus-circle pull-right cmdAction cursor" data-action="remove"></i>';
    tr += '</td>';
    tr += '</tr>';
    $('#table_cmd tbody').append(tr);
    $('#table_cmd tbody tr:last').setValues(_cmd, '.cmdAttr');
    if (isset(_cmd.type)) {
        $('#table_cmd tbody tr:last .cmdAttr[data-l1key=type]').value(init(_cmd.type));
    }
    jeedom.cmd.changeType($('#table_cmd tbody tr:last'), init(_cmd.subType));
}


plugin_info/install.php
<?php

/* This file is part of Jeedom.
 *
 * Jeedom is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jeedom is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jeedom. If not, see <http://www.gnu.org/licenses/>.
 */

require_once dirname(__FILE__) . '/../../../core/php/core.inc.php';

function ree_install() {
    
    
}

function ree_update() {
     
    
}


function ree_remove() {
    
}

?>


core/class/ree.class.php
<?php
/* This file is part of Jeedom.
 *
 * Jeedom is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jeedom is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jeedom. If not, see <http://www.gnu.org/licenses/>.
 */
/* * ***************************Includes********************************* */
require_once __DIR__  . '/../../../../core/php/core.inc.php';
class ree extends eqLogic {
    /*     * *************************Attributs****************************** */
     public static $_horas = array('00h', '01h', '02h', '03h', '04h', '05h', '06h', '07h', '08h', '09h', '10h', '11h', '12h', '13h', '14h', '15h', '16h', '17h', '18h', '19h', '20h', '21h', '22h', '23h');
    /*     * ***********************Methode static*************************** */
    /*
     * Fonction exécutée automatiquement toutes les minutes par Jeedom
      public static function cron() {
      }
     */
    /*
     * Fonction exécutée automatiquement toutes les heures par Jeedom
      public static function cronHourly() {
      }
     */
    /*
     * Fonction exécutée automatiquement tous les jours par Jeedom */
      public static function cron30() {
		 if ($_eqLogic_id == null) { // La fonction n’a pas d’argument donc on recherche tous les équipements du plugin
                        $eqLogics = self::byType('ree', true);
                } else {// La fonction a l’argument id(unique) d’un équipement(eqLogic)
                        $eqLogics = array(self::byId($_eqLogic_id));
                }
                foreach ($eqLogics as $ree) {
                        if ($ree->getIsEnable() == 1) {//vérifie que l'équipement est acitf
                                $cmd = $ree->getCmd(null, 'refresh');//retourne la commande "refresh si elle exxiste
                                if (!is_object($cmd)) {//Si la commande n'existe pas
                                  continue; //continue la boucle
                                }
                                $cmd->execCmd(); // la commande existe on la lance
                        }
                }
      }
    /*     * *********************Méthodes d'instance************************* */
    public function preInsert() {
        
    }
    public function postInsert() {
        
    }
    public function preSave() {
         $this->setDisplay("width","225px");
    }
    public function postSave() {
        $refresh = $this->getCmd(null, 'refresh');
                if (!is_object($refresh)) {
                        $refresh = new reeCmd();
                        $refresh->setName(__('Rafraichir', __FILE__));
                }
                $refresh->setEqLogic_id($this->getId());
                $refresh->setLogicalId('refresh');
                $refresh->setType('action');
                $refresh->setSubType('other');
                $refresh->save();

        for ($x = 0; $x <= 23; $x++) {
                    $info = $this->getCmd(null, self::$_horas[$x]);
                    if (!is_object($info)) {
                            $info = new reeCmd();
                            $info->setName(__(self::$_horas[$x], __FILE__));
                    }
                    $info->setLogicalId(self::$_horas[$x]);
                    $info->setEqLogic_id($this->getId());
                    $info->setType('info');
		    $info->setTemplate('dashboard', 'line');
		    $info->setTemplate('mobile', 'line');
		    $info->setUnite('€/KWh');
		    $info->setSubType('numeric');
                    $info->save();
        }
		    $info = $this->getCmd(null, 'name');
                    if (!is_object($info)) {
                            $info = new reeCmd();
                            $info->setName(__('name', __FILE__));
                    }
                    $info->setLogicalId('name');
                    $info->setEqLogic_id($this->getId());
                    $info->setType('info');
                    $info->setSubType('string');
                    $info->save();
		    $info = $this->getCmd(null, 'date');
                    if (!is_object($info)) {
                            $info = new reeCmd();
                            $info->setName(__('date', __FILE__));
                    }
                    $info->setLogicalId('date');
                    $info->setEqLogic_id($this->getId());
                    $info->setType('info');
                    $info->setSubType('string');
                    $info->save();

			$info1 = $this->getCmd(null, 'hour');
                    if (!is_object($info1)) {
                            $info1 = new reeCmd();
                            $info1->setName(__('hour', __FILE__));
                    }
                    $info1->setLogicalId('hour');
                    $info1->setEqLogic_id($this->getId());
                    $info1->setType('info');
		    $info1->setTemplate('dashboard', 'line');
                    $info1->setTemplate('mobile', 'line');
                    $info1->setUnite('€/KWh');
                    $info1->setSubType('numeric');
                    $info1->save();

		    $info2 = $this->getCmd(null, 'cheaphour');
                    if (!is_object($info2)) {
                            $info2 = new reeCmd();
                            $info2->setName(__('cheaphour', __FILE__));
                    }
                    $info2->setLogicalId('cheaphour');
                    $info2->setEqLogic_id($this->getId());
                    $info2->setType('info');
                    $info2->setSubType('string');
                    $info2->save();

                $info3 = $this->getCmd(null, 'expensivehour');
                if (!is_object($info3)) {
                        $info3 = new reeCmd();
                        $info3->setName(__('expensivehour', __FILE__));
                }
                $info3->setLogicalId('expensivehour');
                $info3->setEqLogic_id($this->getId());
                $info3->setType('info');
                $info3->setSubType('string');
                $info3->save();
	    
	    	//Modificación para añdir gráfica
	    	$info4 = $this->getCmd(null, 'graph');
                if (!is_object($info4)) {
                        $info4 = new reeCmd();
                        $info4->setName(__('graph', __FILE__));
                }
                $info4->setLogicalId('graph');
                $info4->setEqLogic_id($this->getId());
                $info4->setType('info');
	        //$info4->setTemplate('dashboard', 'Ree/Ree');
                $info4->setSubType('string');
                $info4->save();
    }
    public function preUpdate() {
        
    }
	
    public function postUpdate() {
       // self::cronHourly($this->getId());// lance la fonction cronHourly
          $refresh = $this->getCmd(null, 'refresh');
                if (!is_object($refresh)) {
                        $refresh = new reeCmd();
                        $refresh->setName(__('Rafraichir', __FILE__));
                
                $refresh->setEqLogic_id($this->getId());
                $refresh->setLogicalId('refresh');
                $refresh->setType('action');
                $refresh->setSubType('other');
                $refresh->save();}

        for ($x = 0; $x <= 23; $x++) {
                    $info = $this->getCmd(null, self::$_horas[$x]);
                    if (!is_object($info)) {
                            $info = new reeCmd();
                            $info->setName(__(self::$_horas[$x], __FILE__));
                    
                    $info->setLogicalId(self::$_horas[$x]);
                    $info->setEqLogic_id($this->getId());
                    $info->setType('info');
		    $info->setTemplate('dashboard', 'line');                    //$info->setDisplay('title_placeholder', __('Options', __FILE__));
			$info->setTemplate('mobile', 'line');
			$info->setUnite('€/KWh');
			$info->setSubType('numeric');
                    $info->save();}
        }
		    $info = $this->getCmd(null, 'name');
                    if (!is_object($info)) {
                            $info = new reeCmd();
                            $info->setName(__('name', __FILE__));
                    
                    $info->setLogicalId('name');
                    $info->setEqLogic_id($this->getId());
                    $info->setType('info');
                    $info->setSubType('string');
                    $info->save();}
		    
		    $info = $this->getCmd(null, 'date');
                    if (!is_object($info)) {
                            $info = new reeCmd();
                            $info->setName(__('date', __FILE__));
                    
                    $info->setLogicalId('date');
                    $info->setEqLogic_id($this->getId());
                    $info->setType('info');
                    $info->setSubType('string');
                    $info->save();}

			$info1 = $this->getCmd(null, 'hour');
                    if (!is_object($info1)) {
                            $info1 = new reeCmd();
                            $info1->setName(__('hour', __FILE__));
                    
                    $info1->setLogicalId('hour');
                    $info1->setEqLogic_id($this->getId());
                    $info1->setType('info');
                    $info1->setTemplate('dashboard', 'line');
                    $info1->setTemplate('mobile', 'line');
                    $info1->setUnite('€/KWh');
                    $info1->setSubType('numeric');
                    $info1->save();}

		    $info2 = $this->getCmd(null, 'cheaphour');
                    if (!is_object($info2)) {
                            $info2 = new reeCmd();
                            $info2->setName(__('cheaphour', __FILE__));
                    
                    $info2->setLogicalId('cheaphour');
                    $info2->setEqLogic_id($this->getId());
                    $info2->setType('info');
                    $info2->setSubType('string');
                    $info2->save();}

                $info3 = $this->getCmd(null, 'expensivehour');
                if (!is_object($info3)) {
                        $info3 = new reeCmd();
                        $info3->setName(__('expensivehour', __FILE__));
                
                $info3->setLogicalId('expensivehour');
                $info3->setEqLogic_id($this->getId());
                $info3->setType('info');
                $info3->setSubType('string');
                $info3->save();}
	    
	    	//Modificaciín para añadir gráfica
	    	$info4 = $this->getCmd(null, 'graph');
                if (!is_object($info4)) {
                        $info4 = new reeCmd();
                        $info4->setName(__('graph', __FILE__));
                
                $info4->setLogicalId('graph');
                $info4->setEqLogic_id($this->getId());
                $info4->setType('info');
		//$info4->setTemplate('dashboard', 'Ree/Ree');
                $info4->setSubType('string');
                $info4->save();}
		
		self::cron30($this->getId());// lance la fonction cronHourly
    }
    public function preRemove() {
        
    }
    public function postRemove() {
       
    }
    /*
     * Non obligatoire mais permet de modifier l'affichage du widget si vous en avez besoin
      public function toHtml($_version = 'dashboard') {
      }
     */
    /*
     * Non obligatoire mais ca permet de déclencher une action après modification de variable de configuration
    public static function postConfig_<Variable>() {
    }
     */
    /*
     * Non obligatoire mais ca permet de déclencher une action avant modification de variable de configuration
    public static function preConfig_<Variable>() {
    }
     */
    /*     * **********************Getteur Setteur*************************** */
}
class reeCmd extends cmd {
    /*     * *************************Attributs****************************** */
    public static $_horas2 = array('00h', '01h', '02h', '03h', '04h', '05h', '06h', '07h', '08h', '09h', '10h', '11h', '12h', '13h', '14h', '15h', '16h', '17h', '18h', '19h', '20h', '21h', '22h', '23h');
    /*     * ***********************Methode static*************************** */
    /*     * *********************Methode d'instance************************* */
    /*
     * Non obligatoire permet de demander de ne pas supprimer les commandes même si elles ne sont pas dans la nouvelle configuration de l'équipement envoyé en JS
      public function dontRemoveCmd() {
      return true;
      }
     */
    public function execute($_options = array()) {
	        if ($this->getLogicalId() == 'refresh')
	        {
                $fecha_actual = date("Y-m-d");
		//$eqConfig = $this->getConfigKey();
                $token = config::byKey('param1', 'ree');
                $eqLogic = $this->getEqLogic();
		$selhora = $eqLogic->getConfiguration('hora');
                //sumo 1 dÃ­a
                $fecha_siguiente = date("Y-m-d",strtotime($fecha_actual."+ 1 days"));
		$fecha_actual_hora = strtotime(date("Y-m-d H:i:00",time()));
		/*$fecha_entrada_hora = strtotime($fecha_actual . "23:59:00");*/
		$fecha_entrada_hora = strtotime($fecha_actual . $selhora);
		/*$fecha_entrada_hora = strtotime($fecha_actual . "21:59:00");*/
		if($fecha_actual_hora > $fecha_entrada_hora)
		{
			 $fecha_siguiente = date("Y-m-d",strtotime($fecha_actual."+ 1 days"));
		}else
		{
			 $fecha_siguiente = $fecha_actual;
		}
                $ch = curl_init();
                
                //$eqConfig = $this->getConfigKey();
                //$token = config::byKey('param1', 'ree');
                //$eqLogic = $this->getEqLogic();
		$seltaux = $eqLogic->getConfiguration('taux');
                switch ($seltaux) {
 		 case "PVPC":
        		$taux = "1013";
        		break;
    		case "PVPC 2.0 DHA":
        		$taux = "1014";
        		break;
    		case "PVPC 2.0 DHS":
        		$taux = "1015";
        		break;
		}
                $headers = array(
                        "Accept: application/json; application/vnd.esios-api-v1+json",
                        "Content-Type: application/json",
                        "Host: api.esios.ree.es",
                        "Authorization: Token token=" . $token,
                        "Cookie: "
                );
                curl_setopt($ch, CURLOPT_URL,"https://api.esios.ree.es/indicators/" . $taux . "?start_date=" . $fecha_siguiente . "T00:00:00&end_date=" . $fecha_siguiente . "T23:50:00");
                curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
                curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
                $response = curl_exec($ch);
                curl_close($ch);
                $obj=json_decode($response,true);
		// Imprime el JSON en el log para poder consultarlo
		log::add('ree', 'debug', 'Datos json: ' . print_r($response, True));
               // On rÃ©cupÃ¨re l'Ã©quipement Ã  partir de l'identifiant fournit par la commande
               $reeObj = ree::byId($this->getEqlogic_id());
                $graph = "";
		// On rÃ©cupÃ¨re la commande 'data' appartenant Ã  l'Ã©quipement			
               for ($x = 0; $x <= 23; $x++) {
		   $graph = $graph . ($obj['indicator']['values'][$x]['value'] / 1000) . ",";
                   $dataCmd = $reeObj->getCmd('info', self::$_horas2[$x]);
                   $dataCmd->event($obj['indicator']['values'][$x]['value'] / 1000);
                   //$dataCmd->event($globalp1);
                   $dataCmd->save();
			$tarifa[] = $obj['indicator']['values'][$x]['value'] / 1000;
                   }
		//grabamos los datos de gráfico
	    	//$dataGraph->event($graph);
	    	//$dataGraph->save();
               }
		
	    	$dataCmd = $reeObj->getCmd('info', 'graph');
	    	   //$graph = $graph . ($obj['indicator']['short_name']) . "," . substr($obj['indicator']['values'][0]['datetime'], 0, 10);
                   $dataCmd->event($graph);
                   //$dataCmd->event($globalp1);
                   $dataCmd->save();
	    
	    	$dataCmd = $reeObj->getCmd('info', 'name');
                   $dataCmd->event($obj['indicator']['short_name']);
                   //$dataCmd->event($globalp1);
                   $dataCmd->save();

    		 $dataCmd = $reeObj->getCmd('info', 'date');
                   $dataCmd->event(substr($obj['indicator']['values'][0]['datetime'], 0, 10));
                   $dataCmd->save();

		$dt = new DateTime();
		$dataCmd1 = $reeObj->getCmd('info', 'hour');//incluimos un comando con la tarifa de la hora actual
                        $dataCmd1->event($obj['indicator']['values'][(string)$dt->format('G')]['value'] / 1000);
                        //$dataCmd1->event($dt->format('G'));
                        $dataCmd1->save();
                        //$dataCmd->event($obj['indicator']['values'][0]['datetime']);
                        //$dataCmd->save();

		asort($tarifa); //ordenamos las este arriba el primer elecmento sera el mas barato y el ultimo el mas caro
		 reset($tarifa); //nos aseguramos de estar en el primer elemento ya ordenador del array que incluye las tarifas de todas las horas de$
                        //end($tarifa);
                        $dataCmd2 = $reeObj->getCmd('info', 'cheaphour'); //incluimos un elemento en el equipo con la hora mas barata del dia
                        if (key($tarifa) < 10) {
				$cheaphour = "0" . key($tarifa);
			}else 
			{
				$cheaphour = key($tarifa);
			}
			$dataCmd2->event($cheaphour . "h");
                        $dataCmd2->save();

                        reset($tarifa);//nos aseguramos de estar en el primer elemento ya ordenador del array que incluye las tarifas de todas las horas del$
                        end($tarifa);//nos vamos al ultimo elemento del array de tarifas y por tanto a la hora mas cara del dia
                        $dataCmd3 = $reeObj->getCmd('info', 'expensivehour');
                        if (key($tarifa) < 10) {
                                $expensivehour = "0" . key($tarifa);
                        }else
                        {
                                $expensivehour = key($tarifa);
                        }
			$dataCmd3->event($expensivehour . "h");
                        $dataCmd3->save();

    }
    /*     * **********************Getteur Setteur*************************** */
}


plugin_info/configuration.php
<?php
/* This file is part of Jeedom.
 *
 * Jeedom is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jeedom is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jeedom. If not, see <http://www.gnu.org/licenses/>.
 */

require_once dirname(__FILE__) . '/../../../core/php/core.inc.php';
include_file('core', 'authentification', 'php');
if (!isConnect()) {
    include_file('desktop', '404', 'php');
    die();
}
?>
<form class="form-horizontal">
    <fieldset>
        <div class="form-group">
            <label class="col-lg-4 control-label">{{Token}}</label>
            <div class="col-lg-2">
                <input class="configKey form-control" data-l1key="param1" />
            </div>
        </div>
  </fieldset>
</form>



desktop/php/ree.php
<?php
if (!isConnect('admin')) {
	throw new Exception('{{401 - Accès non autorisé}}');
}
$plugin = plugin::byId('ree');
sendVarToJS('eqType', $plugin->getId());
$eqLogics = eqLogic::byType($plugin->getId());
?>

<div class="row row-overflow">
    <div class="col-lg-2 col-md-3 col-sm-4">
        <div class="bs-sidebar">
            <ul id="ul_eqLogic" class="nav nav-list bs-sidenav">
                <a class="btn btn-default eqLogicAction" style="width : 100%;margin-top : 5px;margin-bottom: 5px;" data-action="add"><i class="fa fa-plus-circle"></i> {{Ajouter un équipement}}</a>
                <li class="filter" style="margin-bottom: 5px;"><input class="filter form-control input-sm" placeholder="{{Rechercher}}" style="width: 100%"/></li>
                <?php
foreach ($eqLogics as $eqLogic) {
	$opacity = ($eqLogic->getIsEnable()) ? '' : jeedom::getConfiguration('eqLogic:style:noactive');
	echo '<li class="cursor li_eqLogic" data-eqLogic_id="' . $eqLogic->getId() . '" style="' . $opacity .'"><a>' . $eqLogic->getHumanName(true) . '</a></li>';
}
		    ?>
           </ul>
       </div>
   </div>

   <div class="col-lg-10 col-md-9 col-sm-8 eqLogicThumbnailDisplay" style="border-left: solid 1px #EEE; padding-left: 25px;">
    <legend>{{Mes équipement}}</legend>
  <legend><i class="fa fa-cog"></i>  {{Gestion}}</legend>
  <div class="eqLogicThumbnailContainer">
      <div class="cursor eqLogicAction" data-action="add" style="text-align: center; background-color : #ffffff; height : 120px;margin-bottom : 10px;padding : 5px;border-radius: 2px;width : 160px;margin-left : 10px;" >
        <i class="fa fa-plus-circle" style="font-size : 6em;color:#94ca02;"></i>
        <br>
        <span style="font-size : 1.1em;position:relative; top : 23px;word-break: break-all;white-space: pre-wrap;word-wrap: break-word;color:#94ca02">{{Ajouter}}</span>
    </div>
      <div class="cursor eqLogicAction" data-action="gotoPluginConf" style="text-align: center; background-color : #ffffff; height : 120px;margin-bottom : 10px;padding : 5px;border-radius: 2px;width : 160px;margin-left : 10px;">
      <i class="fa fa-wrench" style="font-size : 6em;color:#767676;"></i>
    <br>
    <span style="font-size : 1.1em;position:relative; top : 15px;word-break: break-all;white-space: pre-wrap;word-wrap: break-word;color:#767676">{{Configuration}}</span>
  </div>
  </div>
  <legend><i class="fa fa-table"></i> {{Mes templates}}</legend>
<div class="eqLogicThumbnailContainer">
    <?php
foreach ($eqLogics as $eqLogic) {
	$opacity = ($eqLogic->getIsEnable()) ? '' : jeedom::getConfiguration('eqLogic:style:noactive');
	echo '<div class="eqLogicDisplayCard cursor" data-eqLogic_id="' . $eqLogic->getId() . '" style="text-align: center; background-color : #ffffff; height : 200px;margin-bottom : 10px;padding : 5px;border-radius: 2px;width : 160px;margin-left : 10px;' . $opacity . '" >';
	echo '<img src="' . $plugin->getPathImgIcon() . '" height="105" width="95" />';
	echo "<br>";
	echo '<span style="font-size : 1.1em;position:relative; top : 15px;word-break: break-all;white-space: pre-wrap;word-wrap: break-word;">' . $eqLogic->getHumanName(true, true) . '</span>';
	echo '</div>';
}
?>
</div>
</div>

<div class="col-lg-10 col-md-9 col-sm-8 eqLogic" style="border-left: solid 1px #EEE; padding-left: 25px;display: none;">
	<a class="btn btn-success eqLogicAction pull-right" data-action="save"><i class="fa fa-check-circle"></i> {{Sauvegarder}}</a>
  <a class="btn btn-danger eqLogicAction pull-right" data-action="remove"><i class="fa fa-minus-circle"></i> {{Supprimer}}</a>
  <a class="btn btn-default eqLogicAction pull-right" data-action="configure"><i class="fa fa-cogs"></i> {{Configuration avancée}}</a>
  <ul class="nav nav-tabs" role="tablist">
    <li role="presentation"><a href="#" class="eqLogicAction" aria-controls="home" role="tab" data-toggle="tab" data-action="returnToThumbnailDisplay"><i class="fa fa-arrow-circle-left"></i></a></li>
    <li role="presentation" class="active"><a href="#eqlogictab" aria-controls="home" role="tab" data-toggle="tab"><i class="fa fa-tachometer"></i> {{Equipement}}</a></li>
    <li role="presentation"><a href="#commandtab" aria-controls="profile" role="tab" data-toggle="tab"><i class="fa fa-list-alt"></i> {{Commandes}}</a></li>
  </ul>
  <div class="tab-content" style="height:calc(100% - 50px);overflow:auto;overflow-x: hidden;">
    <div role="tabpanel" class="tab-pane active" id="eqlogictab">
      <br/>
    <form class="form-horizontal">
        <fieldset>
            <div class="form-group">
                <label class="col-sm-3 control-label">{{Nom de l'équipement}}</label>
                <div class="col-sm-3">
                    <input type="text" class="eqLogicAttr form-control" data-l1key="id" style="display : none;" />
                    <input type="text" class="eqLogicAttr form-control" data-l1key="name" placeholder="{{Nom de l'équipement template}}"/>
                </div>
            </div>
            <div class="form-group">
                <label class="col-sm-3 control-label" >{{Objet parent}}</label>
                <div class="col-sm-3">
                    <select id="sel_object" class="eqLogicAttr form-control" data-l1key="object_id">
                        <option value="">{{Aucun}}</option>
                        <?php
foreach (jeeObject::all() as $object) {
	echo '<option value="' . $object->getId() . '">' . $object->getName() . '</option>';
}
?>
                   </select>
               </div>
           </div>
	   <div class="form-group">
                <label class="col-sm-3 control-label">{{Catégorie}}</label>
                <div class="col-sm-9">
                 <?php
                    foreach (jeedom::getConfiguration('eqLogic:category') as $key => $value) {
                    echo '<label class="checkbox-inline">';
                    echo '<input type="checkbox" class="eqLogicAttr" data-l1key="category" data-l2key="' . $key . '" />' . $value['name'];
                    echo '</label>';
                    }
                  ?>
               </div>
           </div>
	<div class="form-group">
		<label class="col-sm-3 control-label"></label>
		<div class="col-sm-9">
			<label class="checkbox-inline"><input type="checkbox" class="eqLogicAttr" data-l1key="isEnable" checked/>{{Activer}}</label>
			<label class="checkbox-inline"><input type="checkbox" class="eqLogicAttr" data-l1key="isVisible" checked/>{{Visible}}</label>
		</div>
	</div>
    <div class="form-group">

              <label class="col-sm-3 control-label">{{Tarifa}}</label>
              <div class="col-sm-3">
                <select id="sel_taux" class="form-control eqLogicAttr" data-l1key="configuration" data-l2key="taux">
                  <option value="PVPC">{{PVPC}}</option>
                  <option value="PVPC 2.0 DHA">{{PVPC 2.0 DHA}}</option>
                  <option value="PVPC 2.0 DHS">{{PVPC 2.0 DHS}}</option>
		 </select>
              </div>

            </div>
     <div class="form-group">

              <label class="col-sm-3 control-label">{{Hora Cambio Día}}</label>
              <div class="col-sm-3">
                <select id="sel_hora" class="form-control eqLogicAttr" data-l1key="configuration" data-l2key="hora">
                  <option value="20:29:00">{{20:30 horas}}</option>
		  <option value="20:59:00">{{21 horas}}</option>
		  <option value="21:29:00">{{21:30 horas}}</option>
		  <option value="21:59:00">{{22 horas}}</option>
                  <option value="22:29:00">{{22:30 horas}}</option>
		  <option value="22:59:00" selected="selected">{{23 horas}}</option>
                  <option value="23:29:00">{{23:30 horas}}</option>
		  <option value="23:59:00">{{00 horas}}</option>
		 </select>
              </div>

            </div>
</fieldset>
</form>
</div>
      <div role="tabpanel" class="tab-pane" id="commandtab">
<a class="btn btn-success btn-sm cmdAction pull-right" data-action="add" style="margin-top:5px;"><i class="fa fa-plus-circle"></i> {{Commandes}}</a><br/><br/>
<table id="table_cmd" class="table table-bordered table-condensed">
    <thead>
        <tr>
            <th>{{Nom}}</th><th>{{Type}}</th><th>{{Configuration}}</th><th>{{Action}}</th>
        </tr>
    </thead>
    <tbody>
    </tbody>
</table>
</div>
</div>

</div>
</div>

<?php include_file('desktop', 'ree', 'js', 'ree');?>
<?php include_file('core', 'plugin.template', 'js');?>
